mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/vault/home/HomeVaultPresenter.kt

```diff
@@ -0,0 +1,114 @@
+package rs.readahead.washington.mobile.views.fragment.vault.home
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+import com.hzontal.tella_vault.VaultFile
+import com.hzontal.tella_vault.filter.FilterType
+import com.hzontal.tella_vault.filter.Limits
+import com.hzontal.tella_vault.filter.Sort
+import io.reactivex.Completable
+import io.reactivex.CompletableSource
+import io.reactivex.SingleSource
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.functions.Function
+import io.reactivex.schedulers.Schedulers
+import rs.readahead.washington.mobile.BuildConfig
+import rs.readahead.washington.mobile.MyApplication
+import rs.readahead.washington.mobile.data.database.DataSource
+import rs.readahead.washington.mobile.data.database.KeyDataSource
+import rs.readahead.washington.mobile.data.sharedpref.Preferences
+import rs.readahead.washington.mobile.data.sharedpref.SharedPrefs
+import rs.readahead.washington.mobile.media.MediaFileHandler
+
+class HomeVaultPresenter constructor(var view: IHomeVaultPresenter.IView?) :
+    IHomeVaultPresenter.IPresenter {
+    private var keyDataSource: KeyDataSource = MyApplication.getKeyDataSource()
+    private var disposable: CompositeDisposable? = null
```

Comment:
On many other places also - why is this `CompositeDisposable?` and not `CompositeDisposable`. This object need not been created, with `disposable` being null at any point.